compiler/optimizer/optimizer.go

```diff
@@ -322,3 +340,144 @@ func pushDown(trunk *dag.Trunk) {
 	}
 	trunk.Pushdown = filter
 }
+
+// newRangePruner returns a new predicate based on the input predicate pred
+// that when applied to an input value (i.e., "this") with fields from/to, returns
+// true if comparisons in pred against literal values can for certain rule out
+// that pred would be true for any value in the from/to range.  From/to are presumed
+// to be ordered according to the order o.  This is used to prune metadata objects
+// from a scan when we know the pool-key range of the object could not satisfy
```

Comment:
Nit: We don't usually hyphenate this.
```suggestion
// from a scan when we know the pool key range of the object could not satisfy
```

Reply:
I thought compound nouns used as an adjective should be hyphenated but I will change it.